CONTRIBUTING.md

```diff
@@ -0,0 +1,435 @@
+  store itself. It is also clear what exactly is exposed.
+
+- A single context property injected from the backend should be exposed only
+  once in a single store.
+
+  Bad:
+
+  ```qml
+  // SwapStore.qml
+  QtObject {
+    readonly property var flatNetworks: networksModule.flatNetworks
+  }
+
+  // CommunityTokensStore.qml
+  QtObject {
+    readonly property var flatNetworks: networksModule.flatNetworks
+  }
+  ```
+
+  Stores represent the state of the application but they are not aware how that
+  state is rendered and interacted by UI components. There is no rule that a
+  single component must take a single store with everything exposed what is
+  needed for that component. On the contrary, the UI component should take all
+  the stores (assuming that stores do not duplicate exposition) it needs as a
+  dependency. Moving on, in most cases a UI component should only take
+  the models/properties it needs and not the entire store.
+
+- Stores are not singletons (if an existing singleton store needs to be used,
+  still can be taken by a component as an explicit dependency).
+
+  When stores are singleton, UI components can access backend from arbitrary
+  places, in an implicit way. Dependencies are harder to track, APIs are not
+  clear as they could be and it causes other problems covered in the `general` section.
+
+  ```qml
+  isGoerliEnabled: root.rootStore.isGoerliEnabled         // OK
+  isGoerliEnabled: WalletStores.RootStore.isGoerliEnabled // Bad
+  ```
+
+- Exposed properties are read-only, state modification is done by methods.
+
+  This approach makes the data flow unidirectional. UI always transforms and
+  renders read-only data. UI requests any changes via methods (also called actions).
+  Those actions may result in updates of one or more read-only properties/models.
+
+- Stores should be always typed, no need to use `var`.
+
+  Thanks to overriding proper import paths, typed stores are not a problem for
+  tests and Storybook pages. They can be freely mocked, with the type preserved.
+
+  ```qml
+  required property TransactionStore store // OK
+  required property var store              // Bad
+  ```
+
+## Adaptors
+
+Adaptors are special type of `QML` components responsible for transforming backend's data.
+Thanks to isolating data (especially models) transformations in adaptors, views
+take possibly simple, ready to display data (models and other read-only properties).
+
+Adaptors are usually a composition of proxy models (from external library
+`SortFilterProxyModel` and from own library of proxy models: `ObjectProxyModel`,
+`LeftJoinModel`, `GroupingModel`, `ConcatModel`, `RenamingProxyModel`,
+`MovableModel`, `WritableModel`).
+
+Diagram depicting custom proxy models can be found [here](https://miro.com/app/board/uXjVKTuFYOU=/)
+
+- Adaptors are data-oriented and not tightly coupled to a specific view. Adaptors' names
+should reflect the transformation they do instead of following the naming of the UI component
```

Comment:
hmm is this cause they could be reused?I thought adaptors are linked with a certain view, but I guess my understanding is wrong here.

Reply:
In the end that "rule" is mostly about naming. When the adaptor is called after component consuming the adaptor's output model (instead of name reflecting more what it's doing) then it may create confusion when reusing that adaptor for another ui component (name will match first component but not necessarily others) or other adaptors.